OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -80,11 +80,15 @@ public void Killed(Actor self, AttackInfo e)
 			if (Info.UseDeathTypeSuffix)
 			{
 				var warhead = e.Warhead as DamageWarhead;
-				var damageType = warhead.DamageTypes.Intersect(Info.DeathTypes.Keys).FirstOrDefault();
-				if (damageType == null)
+				var wda = self.TraitsImplementing<WithDeathAnimation>().FirstOrDefault(s => s.Info.DeathTypes.Keys.Any(k => warhead.DamageTypes.Contains(k)));
```

Comment:
This feels really ugly, and is now covered by a lint check.  How about we just remove it?